packages/plugins/Scale/src/components/Scale.vue

```diff
@@ -15,12 +32,57 @@
 
 <script lang="ts">
 import Vue from 'vue'
-import { mapGetters } from 'vuex'
+import { mapGetters, mapMutations, mapActions } from 'vuex'
+import thousandsSeparator from '../utils/thousandsSeperator'
+import beautifyScale from '../utils/beautifyScale'
 
 export default Vue.extend({
   name: 'PolarScale',
   computed: {
-    ...mapGetters('plugin/scale', ['scaleToOne', 'scaleWithUnit']),
+    ...mapGetters('plugin/scale', [
+      'scaleToOne',
+      'scaleValue',
+      'scaleWithUnit',
+      'showScaleSwitcher',
+      'zoomOptions',
+    ]),
+    scaleNumberToScale() {
+      return (scale: number) => {
+        return '1 : ' + thousandsSeparator(scale)
+      }
+    },
+    scale: {
+      get() {
+        const scaleToCompare = beautifyScale(this.scaleValue)
+        const bestMatchingScale = this.zoomOptions.reduce((prev, curr) => {
+          return Math.abs(curr.scale - scaleToCompare) <
+            Math.abs(prev.scale - scaleToCompare)
+            ? curr
+            : prev
+        })
+        return bestMatchingScale.scale
+      },
+      set(value: number) {
+        this.setScaleValue(value)
+      },
+    },
+    showSelectOptions() {
+      return this.showScaleSwitcher
+    },
+  },
+  methods: {
+    ...mapMutations('plugin/scale', ['setScaleValue']),
+    ...mapActions('plugin/zoom', ['setZoomLevel']),
+    // TODO: Finding a different solution may be a task to be tackled in the future
+    setZoomLevelByScale(index: number) {
+      if (!this.$store._actions['plugin/zoom/setZoomLevel']) {
```

Comment:
```suggestion
      if (!this.$store.hasModule(['plugin', 'zoom'])) {
```
As implemented in `packages/plugins/Gfi/src/components/Gfi.vue`